readthedocs/projects/models.py

```diff
@@ -480,19 +479,14 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
             if not self.slug:
                 raise Exception(_('Model must have slug'))
         super().save(*args, **kwargs)
-        try:
-            latest = self.versions.filter(slug=LATEST).first()
-            default_branch = self.get_default_branch()
-            if latest and latest.identifier != default_branch:
-                latest.identifier = default_branch
-                latest.save()
-        except Exception:
-            log.exception('Failed to update latest identifier')
+
+        # Update `Version.identifier` for `latest` with the default branch the use has selected,
+        # even if it's `None` (meaning to match the `default_branch` of the repository)
+        self.versions.filter(slug=LATEST).update(identifier=self.default_branch)
```

Comment:
```suggestion
        # Update `Version.identifier` for `latest` with the default branch the user has selected,
        # even if it's `None` (meaning to match the `default_branch` of the repository)
        log.debug("Updating default branch." slug=LATEST, identifier=self.default_branch)
        self.versions.filter(slug=LATEST).update(identifier=self.default_branch)
```

Can the user select `None`, or will it only present them with a list of existing branches? Also probably useful to log this?

Reply:
> Can the user select None, or will it only present them with a list of existing branches?

Yes. The use can select `None` by selecting `-----` from the dropdown.

![Screenshot_2023-01-23_11-24-13](https://user-images.githubusercontent.com/244656/214016772-bd565ce8-7b17-4a1e-9fb2-cb63fd882e4e.png)


> Also probably useful to log this?

I'm not sure to understand what you want to log here. Just changing `log.debug` by `log.info`? Logging something else?

[...]

Oh, I re-read the suggestion: **you** added that `log.debug` line 🙃 . OK, great!